open-defi-api.yaml

```diff
@@ -1472,6 +1516,26 @@ components:
           type: object
           $ref: '#/components/schemas/TimeSpanTradeData'
 
+    TokenOnChainInfo:
+      description: >
+        Basic info about a token from on-chain data.
+      type: object
+      properties:
+        name:
+          type: string
+        symbol:
+          type: string
+        total_supply:
+          type: integer
```

Comment:
There are two ways to express the total supply. More generic amount in decimal format or "raw" amount that is fixed point integer directly from EVM uint256 word.

Let's use  decimal. Some tokens have fractional total supply up to 18 decimals .e.g.

`123000.8712489147123944`

We can use `type: string` type and then apply `format: decimal` for it.

Also let's make sure the backend converts this to decimal format when exporting.

The conversion can be done like this:

```python
total_supply_decimal = Decimal(total_supply_raw) / Decimal(10**decimal_places)

# decimal_places = 18 for most of the tokens
```

Reply:
Makes sense, I saw `uint256` and just used integer with a >=0 constraint, but since fractional supply is actually possible, Decimal sounds better.

Reply:
@miohtama Done. Added the new fields and changed the type of the total supply to a decimal string (and made a proper conversion in the backend endpoint).

Will add the backend logic for the new slug fields in a separate PR.